components/script/dom/htmlimageelement.rs

```diff
@@ -435,14 +435,24 @@ impl VirtualMethods for HTMLImageElement {
 }
 
 fn image_dimension_setter(element: &Element, attr: Atom, value: u32) {
+    use app_units::AU_PER_PX;
+    use std::i32;
+
     // This setter is a bit weird: the IDL type is unsigned long, but it's parsed as
     // a dimension for rendering.
     let value = if value > UNSIGNED_LONG_MAX {
         0
     } else {
         value
     };
-    let dim = LengthOrPercentageOrAuto::Length(Au::from_px(value as i32));
+
+    let pixel_value = if value > (i32::MAX / AU_PER_PX) as u32 {
```

Comment:
I wonder if this belongs in Au::from_px, or whether Au::from_px should return an Option value instead? I don't think that I'm a good person to make that call.